### PR TITLE
dsumaroka/spring-boot-data-jpa

### PR DESCRIPTION
**Pull request by dsumaroka**

### CODE REVIEW

File: src/main/java/springboot/web/example/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +34,15 @@ public BookDto findById(@PathVariable Long id) {
...
     }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookBuId(@PathVariable Long id,
+                                  @RequestBody CreateBookRequestDto createBookRequestDto) {
+        return bookService.updateBookById(id, createBookRequestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteById(@PathVariable Long id) {
```
Comment (liliia-ponomarenko) [inline src/main/java/springboot/web/example/controller/BookController.java:48]: Better add ResponseStatus.NO_CONTENT

File: src/main/java/springboot/web/example/model/Book.java [hunk 2]
```diff
@@ -8,26 +8,31 @@
...
-    @Column(unique = true)
+    @Column(nullable = false, unique = true)
     private String isbn;
-    @NonNull
+    @Column(nullable = false)
     private BigDecimal price;
     private String description;
+    @Column(name = "cover_image")
     private String coverImage;
+    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT")
```
Comment (liliia-ponomarenko) [inline src/main/java/springboot/web/example/model/Book.java]:
```suggestion
    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT(1)")
```

File: src/main/java/springboot/web/example/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -37,4 +37,18 @@ public BookDto findById(Long id) {
                 () -> new EntityNotFoundException("Can't find in DB book with id: " + id));
         return bookMapper.toBookDto(book);
     }
+
+    @Override
+    public BookDto updateBookById(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find in DB book with id: " + id));
```
Comment (liliia-ponomarenko) [inline src/main/java/springboot/web/example/service/impl/BookServiceImpl.java]:
```suggestion
                () -> new EntityNotFoundException("Can't find book with id: " + id));
```

File: src/main/resources/db/changelog/changes/01-create-table-books.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                  type: varchar(255)
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/changes/01-create-table-books.yaml]: Add one empty line to the end of the file;)

**Review by liliia-ponomarenko — APPROVED** [review]
Great! A few minor comments. But please fix it before merging ;)

**Comment by dsumaroka** [comment]
Thank’s) Ok

Вт, 1 окт. 2024 г. в 11:20, Liliia Ponomarenko ***@***.***>:

**Pull request merged**